### PR TITLE
Feat: Support the '_dbt_max_partition' variable for BigQuery dbt projects

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

**Review by tobymao — APPROVED** [review]

File: sqlmesh/dbt/model.py [hunk 1]
```diff
@@ -268,12 +268,61 @@ def to_sqlmesh(self, context: DbtContext) -> Model:
...
+            or self.model_materialization != Materialization.INCREMENTAL
+        ):
+            return None
+
+        data_type = self.partition_by["data_type"]
+        granularity = self.partition_by["granularity"]
+
+        parse_fun = f"parse_{data_type}" if data_type in ("date", "datetime", "timestamp") else None
+        if parse_fun:
+            if granularity not in GRANULARITY_TO_PARTITION_FORMAT:
```
Comment (crericha) [inline sqlmesh/dbt/model.py:301]: Can we validate this in the validator instead?

File: sqlmesh/dbt/model.py [hunk 2]
```diff
@@ -268,12 +268,61 @@ def to_sqlmesh(self, context: DbtContext) -> Model:
...
             query,
             dialect=dialect,
             kind=self.model_kind(context.target),
             start=self.start,
             **optional_kwargs,
-            **self.sqlmesh_model_kwargs(context),
+            **model_kwargs,
         )
+
+    def _dbt_max_partition_blob(self) -> t.Optional[str]:
```
Comment (crericha) [inline sqlmesh/dbt/model.py:289]: Consider adding a comment that this is bigquery only.

File: sqlmesh/dbt/adapter.py [hunk 3]
```diff
@@ -276,6 +290,20 @@ def execute(
             return AdapterResponse("Success"), pandas_to_agate(resp)
         return AdapterResponse("Success"), empty_table()
 
+    def resolve_schema(self, relation: BaseRelation) -> t.Optional[str]:
+        schema = self._map_table_name(relation.database, relation.schema, relation.identifier).db
+        if not schema:
+            return None
```
Comment (crericha) [inline sqlmesh/dbt/adapter.py:296]: Is it valid to get None back? If not, should we raise instead? Same with resolve_identifier.

Comment (izeigerman) [reply sqlmesh/dbt/adapter.py:296]: It is. It matches the behavior of the relation itself.

**Review by crericha — APPROVED** [review]

**Pull request merged**